ipn/ipnlocal/local.go

```diff
@@ -803,6 +803,16 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 			b.logf("[v1] TKA sync error: %v", err)
 		}
 		b.mu.Lock()
+		if b.tka != nil {
+			head, err := b.tka.authority.Head().MarshalText()
+			if err != nil {
+				b.logf("[v1] error marshalling tka head: %v", err)
+			} else {
+				b.cc.SetTKAHead(string(head))
```

Comment:
should you be calling these with a lock held?

do those acquire locks themselves?

Reply:
Control client has its own independent locking, which is acquired and dropped as part of the `SetTKAHead` call.

I can see this being a problem if ControlClient holds its own lock while it does a `setClientStatus` call, otherwise it should be fine?

Reply:
Afaict theres no lock held during the callback:

```go
c.mu.Lock()
c.expiry = &nm.Expiry
c.mu.Unlock()

cb(nm)
```